packages/zod/src/index.ts

```diff
@@ -130,6 +133,40 @@ export const DateFromISOString = z
   )
   .transform((s) => new Date(s));
 
+/**
+ * A Zod schema for Temporal.Instant objects.
+ *
+ * We handle three formats based on our needs:
+ *  - `Date` - dates returned by the pg driver will be a `Date` object
+ *  - `Temporal.Instant` - For idempotence
+ *  - `string` - ISO date strings in JSON
```

Comment:
Apparently the format from the database is already `Date`, that was what was causing the failing tests. We don't use this schema for JSON AFAIK either -- I'm questioning the purpose of `DateFromISOString`.

Reply:
Going to change this in [a8c4d7e](https://github.com/PrairieLearn/PrairieLearn/pull/13059/commits/a8c4d7e273f378849ed482d30ae5afc3104ca5db) so we can discuss that further.